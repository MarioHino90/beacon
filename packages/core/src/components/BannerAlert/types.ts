import React from "react";
import { ThemedStyledProps } from "styled-components";

import { UntransitionedBannerAlertProps } from "../UntransitionedBannerAlert";

export type Intent = "success" | "warning" | "danger" | "info";

export interface BannerAlertProps
  extends UntransitionedBannerAlertProps,
    Omit<StyledBannerTransitionProps, "children"> {}

export interface StyledBannerTransitionProps
  extends React.HTMLAttributes<HTMLDivElement> {
  isOpen: boolean;
}

export interface IntentContextProviderProps {
  intent: Intent;
}

// this uses <any, any> because the extra props in the themed styled component
// do not matter, just that it is themed
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export interface IntentColorArgs extends ThemedStyledProps<any, any> {
  intent: Intent;
}
