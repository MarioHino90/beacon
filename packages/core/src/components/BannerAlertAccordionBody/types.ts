export interface BannerAlertAccordionBodyProps
  extends React.HTMLAttributes<HTMLDivElement> {}
