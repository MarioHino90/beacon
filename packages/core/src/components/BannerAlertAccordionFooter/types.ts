export interface BannerAlertAccordionFooterProps
  extends React.HTMLAttributes<HTMLDivElement> {}
