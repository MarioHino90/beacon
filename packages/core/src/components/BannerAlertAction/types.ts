import { ButtonProps } from "../Button";
import { Intent } from "../BannerAlert";

export interface BannerAlertActionProps
  extends Omit<ButtonProps, "variant" | "intent"> {
  /** * the visual intent */
  intent?: Intent;
}
