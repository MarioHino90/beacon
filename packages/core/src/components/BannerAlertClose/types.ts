import { Intent } from "../BannerAlert";

export interface BannerAlertCloseProps {
  /**
   * visual banner intents */
  intent: Intent;
  /**
   * on close callack */
  onClose: () => void;
}

export interface StyledProps {
  /**
   * visual banner intents */
  intent: Intent;
}
