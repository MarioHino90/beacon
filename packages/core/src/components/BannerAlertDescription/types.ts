export interface BannerAlertDescriptionProps
  extends React.HTMLAttributes<HTMLDivElement> {}
