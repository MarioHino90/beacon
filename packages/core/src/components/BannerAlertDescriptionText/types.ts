export interface BannerAlertDescriptionTextProps
  extends React.HTMLAttributes<HTMLSpanElement> {}
