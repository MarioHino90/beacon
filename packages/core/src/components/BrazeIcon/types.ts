import React from "react";

export type GenericSpotIcon<Props> = React.LazyExoticComponent<{
  (props: Props): React.ReactElement;
  displayName: string;
}>;

export interface SpotIcons {
  "add-document": GenericSpotIcon<AddDocumentProps>;
  "add-image-or-document": GenericSpotIcon<AddImageOrDocumentProps>;
  "arrow-toggle": GenericSpotIcon<ArrowToggleProps>;
  "campaign-comparison": GenericSpotIcon<CampaignComparisonProps>;
  content: GenericSpotIcon<ContentProps>;
  "email-settings": GenericSpotIcon<EmailSettingsProps>;
  "empty-table": GenericSpotIcon<AddDocumentProps>;
  "funnel-report": GenericSpotIcon<FunnelReportProps>;
  mms: GenericSpotIcon<MMSProps>;
  preview: GenericSpotIcon<PreviewProps>;
  "retention-report": GenericSpotIcon<RetentionReportProps>;
  sms: GenericSpotIcon<SMSProps>;
}

export interface BrazeIconProps extends StyledBrazeIconProps {
  /**
   * icon to be displayed to user */
  name?: keyof SpotIcons;
}

export interface StyledBrazeIconProps
  extends React.HTMLAttributes<HTMLElement> {
  /**
   * child elements to be rendered inside the component tree */
  children?: React.ReactNode | React.ReactNode[];
}

export interface AddDocumentProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface AddImageOrDocumentProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface ArrowToggleProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface CampaignComparisonProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface ContentProps extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface FunnelReportProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface EmailSettingsProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface PreviewProps extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface MMSProps extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface RetentionReportProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface SMSProps extends React.SVGAttributes<HTMLOrSVGElement> {}
