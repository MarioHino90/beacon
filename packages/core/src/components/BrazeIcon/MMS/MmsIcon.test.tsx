import React from "react";
import { shallow } from "enzyme";
import faker from "@faker-js/faker";

import { MMSProps } from "../types";

import { MMS } from "./MMS";

faker.seed(1604940599917);

describe("MMS", () => {
  let props: MMSProps;

  beforeEach(() => {
    props = {
      children: faker.random.words(),
      className: faker.random.word(),
      id: faker.random.word(),
    };
  });

  it("should match the snapshot", () => {
    const wrapper = shallow(<MMS {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
