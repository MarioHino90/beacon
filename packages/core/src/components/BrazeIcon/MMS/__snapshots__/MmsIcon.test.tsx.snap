// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MMS should match the snapshot 1`] = `
<svg
  className="bcl-mms-icon Faso"
  fill="none"
  height="24"
  id="index"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <rect
    height="96"
    transform="translate(-20 -43)"
    width="185"
  />
  <path
    d="M13.791 0.709999H2.64605C2.20639 0.709999 1.84998 1.06641 1.84998 1.50607V22.2039C1.84998 22.6436 2.20639 23 2.64605 23H13.791C14.2307 23 14.5871 22.6436 14.5871 22.2039V1.50607C14.5871 1.06641 14.2307 0.709999 13.791 0.709999Z"
    fill="white"
  />
  <rect
    height="22"
    stroke="black"
    strokeLinejoin="round"
    width="12"
    x="2"
    y="1"
  />
  <path
    d="M2.30774 3.61168H14.2488"
    stroke="black"
    strokeMiterlimit="10"
  />
  <path
    d="M2.30774 18.737H14.2488"
    stroke="black"
    strokeMiterlimit="10"
  />
  <path
    d="M8.21856 21.6921C8.65821 21.6921 9.01463 21.3357 9.01463 20.8961C9.01463 20.4564 8.65821 20.1 8.21856 20.1C7.7789 20.1 7.42249 20.4564 7.42249 20.8961C7.42249 21.3357 7.7789 21.6921 8.21856 21.6921Z"
    fill="black"
  />
  <path
    d="M9.07428 5.9999H22.2095V14.2094H9.07428V5.9999Z"
    fill="white"
  />
  <path
    clipRule="evenodd"
    d="M22 6H9V15H11.6912L11.6913 17.0827L14.7019 15H22V6Z"
    fill="#C4F0F5"
    fillRule="evenodd"
  />
  <circle
    cx="12.2801"
    cy="8.32"
    fill="black"
    r="0.72"
  />
  <path
    d="M15.25 11.38L13 9.57999L11.2 12.73H20.2L17.5 8.22999L15.25 11.38Z"
    fill="black"
  />
  <path
    d="M9.07428 5.9999V5.4999C8.79814 5.4999 8.57428 5.72375 8.57428 5.9999H9.07428ZM22.2095 5.9999H22.7095C22.7095 5.72375 22.4856 5.4999 22.2095 5.4999V5.9999ZM9.07428 14.6199H8.57428C8.57428 14.896 8.79814 15.1199 9.07428 15.1199V14.6199ZM11.6912 14.6199L12.1912 14.6199C12.1912 14.3437 11.9674 14.1199 11.6912 14.1199V14.6199ZM11.6913 17.0827L11.1913 17.0827C11.1913 17.2686 11.2944 17.4392 11.4591 17.5256C11.6238 17.6119 11.8228 17.5997 11.9757 17.4939L11.6913 17.0827ZM15.2514 14.6199V14.1199C15.1497 14.1199 15.0505 14.1508 14.9669 14.2087L15.2514 14.6199ZM22.2095 14.6199V15.1199C22.4856 15.1199 22.7095 14.896 22.7095 14.6199H22.2095ZM9.07428 6.4999H22.2095V5.4999H9.07428V6.4999ZM9.57428 14.6199V5.9999H8.57428V14.6199H9.57428ZM11.6912 14.1199H9.07428V15.1199H11.6912V14.1199ZM12.1913 17.0827L12.1912 14.6199L11.1912 14.6199L11.1913 17.0827L12.1913 17.0827ZM14.9669 14.2087L11.4068 16.6715L11.9757 17.4939L15.5358 15.0311L14.9669 14.2087ZM22.2095 14.1199H15.2514V15.1199H22.2095V14.1199ZM21.7095 5.9999V14.6199H22.7095V5.9999H21.7095Z"
    fill="black"
  />
</svg>
`;
